src/main/java/eu/dissco/core/digitalspecimenprocessor/service/ProcessingService.java

```diff
@@ -256,54 +288,31 @@ private void rollbackUpdatedSpecimen(UpdatedDigitalSpecimenRecord updatedDigital
       }
     }
     rollBackToEarlierVersion(updatedDigitalSpecimenRecord.currentDigitalSpecimen());
-    if (handleNeedsUpdate(updatedDigitalSpecimenRecord.currentDigitalSpecimen().digitalSpecimen(),
-        updatedDigitalSpecimenRecord.digitalSpecimenRecord()
-            .digitalSpecimen())) {
-      fdoRecordBuilder.deleteVersion(updatedDigitalSpecimenRecord.currentDigitalSpecimen());
-    }
     try {
       kafkaService.deadLetterEvent(
           new DigitalSpecimenEvent(updatedDigitalSpecimenRecord.enrichment(),
               updatedDigitalSpecimenRecord.digitalSpecimenRecord()
                   .digitalSpecimen()));
     } catch (JsonProcessingException e) {
-      log.error("Fatal exception, unable to dead letter queue: "
-          + updatedDigitalSpecimenRecord.digitalSpecimenRecord().id(), e);
+      log.error(DLQ_FAILED + updatedDigitalSpecimenRecord.digitalSpecimenRecord().id(), e);
     }
   }
 
   private void rollBackToEarlierVersion(DigitalSpecimenRecord currentDigitalSpecimen) {
     repository.createDigitalSpecimenRecord(List.of(currentDigitalSpecimen));
   }
 
-  private boolean handleNeedsUpdate(DigitalSpecimen currentDigitalSpecimen,
-      DigitalSpecimen digitalSpecimen) {
-    return !currentDigitalSpecimen.physicalSpecimenId().equals(digitalSpecimen.physicalSpecimenId())
-        || isUnEqual(currentDigitalSpecimen, digitalSpecimen, "ods:organisationId")
-        || isUnEqual(currentDigitalSpecimen, digitalSpecimen, "ods:specimenName")
-        || isUnEqual(currentDigitalSpecimen, digitalSpecimen, "ods:physicalSpecimenIdType")
-        || isUnEqual(currentDigitalSpecimen, digitalSpecimen, "ods:physicalSpecimenCollection")
-        || isUnEqual(currentDigitalSpecimen, digitalSpecimen, "dwc:typeStatus")
-        || isUnEqual(currentDigitalSpecimen, digitalSpecimen, "dwca:id");
-  }
-
-  private boolean isUnEqual(DigitalSpecimen currentDigitalSpecimen, DigitalSpecimen digitalSpecimen,
-      String fieldName) {
-    return !Objects.equals(getValueFromAttributes(currentDigitalSpecimen, fieldName),
-        getValueFromAttributes(digitalSpecimen, fieldName));
-  }
-
-  private String getValueFromAttributes(DigitalSpecimen digitalSpecimen, String fieldName) {
-    if (digitalSpecimen.attributes().get(fieldName) != null) {
-      return digitalSpecimen.attributes().get(fieldName).asText();
-    } else {
-      return null;
-    }
-  }
 
   private Set<DigitalSpecimenRecord> createNewDigitalSpecimen(List<DigitalSpecimenEvent> events) {
+    Map<String, String> pidMap;
+    try {
+      pidMap = createNewPidRecords(events);
+    } catch (PidAuthenticationException | PidCreationException e){
+      log.error("Unable to create PID. {}", e.getMessage());
+      return Collections.emptySet();
```

Comment:
Feels like we have to DQL things here if they fail. What happens to the messages if the pidMap comes back empty?

Reply:
Good catch. Workflow is now:
1. create pidmap
2. if operation throws exception, dlq events one by one
3. if a dlq fails, add the event to faileDlqList
4. if our failedDlqList is not empty, log those failed events

pidMap should never be empty. It will either contain correct number of handles or it will throw an exception.